flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -1278,7 +1278,11 @@ object ScalarOperatorGens {
       val falseAction = generateIfElse(ctx, operands, resultType, i + 2)
 
       val Seq(resultTerm, nullTerm) = newNames("result", "isNull")
-      val resultTypeTerm = primitiveTypeTermForType(resultType)
+      val resultTypeTerm = if (trueAction.resultType.isNullable || falseAction.resultType.isNullable) {
+        boxedTypeTermForType(resultType)
```

Comment:
You should not modify here. We always use `primitiveTypeTermForType` as much as possible.